homework-16

Tmsyt-tm opened the pull request:
更新処理機能の実装
追加
・更新用のhtmlを作成
・Repositoryクラス、Controllerクラス、serviceクラスに更新用のメソッドを作成
問題
DBのstudent_courseにデータが無い場合の処理が未実装
updateではなくinsertを使ってデータを新たに追加する処理を考えている。

Tmsyt-tm committed:
入力データが名前欄に全て入る問題を解決
原因：thymeleafの各データの指定を全て名前欄の物にしていた。恐らくコピペして変更を忘れていたと考えられる。

Tmsyt-tm committed:
htmlで受講コースが表示されない問題を解決

Tmsyt-tm committed:
更新処理機能の実装
追加
・更新用のhtmlを作成
・Repositoryクラス、Controllerクラス、serviceクラスに更新用のメソッドを作成
問題
DBのstudent_courseにデータが無い場合の処理が未実装
updateではなくinsertを使ってデータを新たに追加する処理を考えている。

TomooAkamatsu commented on src/main/java/raisentech/Student/Management/StudentRepository.java:
```diff
@@ -1,43 +1,53 @@
-  @Select("SELECT * FROM student_courses")
-  List<StudentsCourses> searchStudentCourses();
+  //@Select("SELECT * FROM student_courses")
+  //List<StudentsCourses> StudentCourses();
```
不要なコメントは削除しましょう。

TomooAkamatsu commented on src/main/java/raisentech/Student/Management/service/StudentService.java:
```diff
@@ -27,46 +27,79 @@ public StudentService(StudentRepository repository, StudentConverter converter)
+    List<StudentsCourses> studentsCourses = repository.findStudentCourses(student.getId());
+    StudentDetail studentDetail = new StudentDetail();
+    studentDetail.setStudent(student);
+    studentDetail.setStudentsCourses(studentsCourses);
```
コンストラクタを使ってsetしたいです。

Tmsyt-tm committed:
不要なコメントの削除とコンストラクタの追加を行いました

変更内容
StudentDetail.javaにコンストラクタを作成

追加したコンストラクタに合わせて以下のメソッドを修正
public List<StudentDetail> convertStudentDetails
public StudentDetail findStudent
public String newStudent
